azure-quantum/azure/quantum/target/target.py

```diff
@@ -132,7 +132,22 @@ def submit(
         :return: Azure Quantum job
         :rtype: Job
         """
+
         input_params = input_params or {}
+
+        # If the input_data is `QSharpCallable` (coming from the IQ# `qsharp` Python Package)
+        # we need to convert it to QIR bitcode and set the necessary parameters for a QIR job.
+        if (input_data and type(input_data).__name__ == "QSharpCallable"):
+            kwargs["input_data_format"] = kwargs.get("input_data_format", "qir.v1")
+            kwargs["content_type"] = kwargs.get("content_type", "qir.v1")
+            kwargs["output_data_format"] = kwargs.get("output_data_format", "microsoft.quantum-results.v1")
```

Comment:
NIT: Adding to kwargs makes it a bit hard to read IMO. It may be better to set input_data_format, output_data_format and content_type directly in lines 141-143, and wrap lines 151-153 within an else block for non-Q#callable paths.

Reply:
Fixed.